### PR TITLE
Correctly check against multiple prefixes

### PR DESCRIPTION
**Pull request by domachine**
This addresses the case described here: https://github.com/secvisogram/secvisogram/issues/200#issuecomment-997767861

### CODE REVIEW

**Review by wrtlbrmpft — APPROVED** [review]

File: app/lib/shared/bcpLanguageTagChecker.js [hunk 1]
```diff
@@ -53,7 +57,11 @@ export default function (tag) {
         (s) =>
           s.subtag.toLowerCase() === variant.toLowerCase() &&
           s.type === 'variant' &&
-          s.prefix === parsed.langtag.language.language?.toLowerCase()
+          s.prefix.includes(
+            /** @type {string} */ (
+              parsed.langtag.language.language?.toLowerCase()
+            )
+          )
```
Comment (tschmidtb51) [inline app/lib/shared/bcpLanguageTagChecker.js:64]: Checking the language is not enough as a prefix might contain other parts. Valid is also:
```
de-CH-1996
pt-BR-abl1943
ja-Latn-hepburn-heploc
sl-rozaj-1994
sl-rozaj-solba-1994
sl-rozaj-bsike-1994
```

Please add them to the tests.

Here is an idea I came up with: Test against the start of the original tag. Feel free to edit that if there is a more elegant way...

```suggestion
          s.prefix.some((p) =>
            tag
              .substring(
                 0,
                 tag.toLowerCase().indexOf(variant.toLowerCase()) - 1
               )
               .toLowerCase()
               .startsWith(p.toLowerCase())
          )
```

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
I found one minor issue here and suggested a code change. I also added some examples that should pass.

**Comment by domachine** [comment]
Okay, I updated the code. I'm not sure, does that also affect the `extlang` subtag?

**Comment by tschmidtb51** [comment]
> Okay, I updated the code. I'm not sure, does that also affect the `extlang` subtag?

@domachine You're right! [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt) states in Section 2.2.2. Extended Language Subtags:

> [...]
> 2.  Extended language subtag records MUST include **exactly one
>         'Prefix' field** indicating an appropriate **subtag or sequence of
>          subtags** for that extended language subtag.

**Comment by domachine** [comment]
Alright, I adapted the code for the extlang check. Could you recheck?

**Comment by tschmidtb51** [comment]
> Alright, I adapted the code for the extlang check. Could you recheck?

On my way.

**Review by tschmidtb51 — APPROVED** [review]
LGTM. Happy to get that merged. I discovered a minor bug (#227) in the overall implementation. However, that is not really related to this PR.

**Comment by tschmidtb51** [comment]
@domachine Just added the fix for #227 as #228.

**Pull request merged**